rust/Cargo.toml

```diff
@@ -9,3 +9,30 @@ inc = { path = "inc", version = "0.1.0" }
 
 [lib]
 crate-type = ["staticlib"]
+
+[profile.rust-dev]
+inherits = "dev"
+opt-level = 2
+debug = false
+overflow-checks = false
+debug-assertions = false
+strip = "symbols"
+
+[profile.rust-debug]
+inherits = "dev"
+opt-level = 1
+incremental = false
+
+[profile.rust-sanitize]
+inherits = "dev"
+opt-level = "s"
```

Comment:
What does "opt-level=s" do?

Reply:
binary size: https://doc.rust-lang.org/cargo/reference/profiles.html

Reply:
Why does "sanitize" mode specifically needs to optimize for size? I'm not saying it's bad, it just sounds strange.

Reply:
The cpp files are compiled with the `-Os` flag in the sanitize mode, so I suggested using an equivalent flag for rust files.

As for why we are optimizing for size in the cpp files - I'm not sure. The original patchset which introduces sanitize mode (https://groups.google.com/g/scylladb-dev/c/wQ_pw5zkJIg/m/bHW-oeO3AwAJ) doesn't seem to explain it, and the author doesn't work with us anymore. If you have a good argument not to optimize for size here, then I'm completely fine with using a different mode.

Reply:
No, I'm fine with this optimization mode. Any other choice will seem equality arbitrary...